src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/ProjectAssetFileWatcher.cs

```diff
@@ -48,19 +70,22 @@ internal class ProjectAssetFileWatcher : OnceInitializedOnceDisposed, IVsFileCha
         /// <summary>
         /// Called on project load.
         /// </summary>
-        [ConfiguredProjectAutoLoad(RequiresUIThread = true)]
+        [ConfiguredProjectAutoLoad(RequiresUIThread = false)]
         [AppliesTo(ProjectCapability.CSharpOrVisualBasicOrFSharp)]
         internal void Load()
         {
-            EnsureInitialized();
+            InitializeAsync();
         }
 
         /// <summary>
         /// Initialize the watcher.
         /// </summary>
-        protected override void Initialize()
+        protected override async Task InitializeCoreAsync(CancellationToken cancellationToken)
         {
-            _fileChangeService = _serviceProvider.GetService<IVsFileChangeEx, SVsFileChangeEx>();
+            _fileChangeService = (IVsFileChangeEx)(await _asyncServiceProvider.GetServiceAsync(typeof(SVsFileChangeEx)).ConfigureAwait(false));
+
+            // Explicitly get back to the thread pool for the rest of this method so we don't tie up the UI thread;
+            await TaskScheduler.Default;
 
             // The tree source to get changes to the tree so that we can identify when the assets file changes.
             var treeSource = _fileSystemTreeProvider.Tree.SyncLinkOptions();
```

Comment:
Since your code no longer blocks the project to load, can you put the code (after the await TaskSchedule.Default into a LoadedProjectAsync block, just to make sure the project won't be unloaded in the middle?  (in which case, some of the code may throw, because MEF may start to be brought down at that point